src/App.tsx

```diff
@@ -17,18 +77,33 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                status={filterStatus}
+                searchTerm={searchTerm}
+                onFilterChange={handleFilterChange}
+                onSearch={handleSearch}
+                onClearSearch={handleClearSearch}
+              />
             </div>
 
-            <div className="block">
+            {isLoading ? (
               <Loader />
-              <TodoList />
-            </div>
+            ) : (
+              <div className="block">
+                <TodoList
+                  selectedItem={selectedTodo}
+                  todos={filteredTodos} // Відфільтровані завдання
+                  onTodoClick={onOpenModal}
+                />
```

Comment:
Check that the `TodoList` component is correctly implemented and imported. Ensure it handles the props `selectedItem`, `todos`, and `onTodoClick` as expected.